src/sage/libs/giac/__init__.py

```diff
@@ -1,348 +1,3 @@
-        sage: B = gb_giac(Cyclic(P))
-        ...
-        sage: B.is_groebner(), B.ideal() == Cyclic(P)
-        (True, True)
-        sage: P = P.change_ring(order='degrevlex(2),degrevlex(2)')
-        sage: B = gb_giac(Cyclic(P))
-        ...
-        sage: B.is_groebner(), B.ideal() == Cyclic(P)
-        (True, True)
-    """
-    try:
-        iter(gens)
-    except TypeError:
-        gens = gens.gens()
-
-    # get the ring from gens
-    P = next(iter(gens)).parent()
-    K = P.base_ring()
-    p = K.characteristic()
-
-    # check if the ideal is zero. (giac 1.2.0.19 segfault)
-    from sage.rings.ideal import Ideal
-    if (Ideal(gens)).is_zero():
-        return PolynomialSequence([P(0)], P, immutable=True)
-
-    # check for name confusions
-    blackgiacconstants = ['i', 'e']  # NB e^k is expanded to exp(k)
-    blacklist = blackgiacconstants + [str(j) for j in libgiac.VARS()]
-    problematicnames = sorted(set(P.gens_dict()).intersection(blacklist))
-
-    if problematicnames:
-        raise ValueError("Variables names %s conflict in giac. Change them or purge them from in giac with libgiac.purge(\'%s\')"
-                         % (problematicnames, problematicnames[0]))
-
-    if K.is_prime_field() and p == 0:
-        F = libgiac(gens)
-    elif K.is_prime_field() and p < 2**31:
-        F = (libgiac(gens) % p)
-    else:
-        raise NotImplementedError("Only prime fields of cardinal < 2^31 are implemented in Giac for Groebner bases.")
-
-    # proof or probabilistic reconstruction
-    if proba_epsilon is None:
-        if proof_polynomial():
-            giacsettings.proba_epsilon = 0
-        else:
-            giacsettings.proba_epsilon = 1e-15
-    else:
-        giacsettings.proba_epsilon = proba_epsilon
-
-    # prot
-    if prot:
-        libgiac('debug_infolevel(2)')
-
-    # threads
-    if threads is not None:
-        giacsettings.threads = threads
-
-    if elim_variables is None:
-        var_names = P.variable_names()
-        order_name = P.term_order().name()
-        if order_name == "degrevlex":
-            giac_order = "revlex"
-        elif order_name == "lex":
-            giac_order = "plex"
-        elif order_name == "deglex":
-            giac_order = "tdeg"
-        else:
-            blocks = P.term_order().blocks()
-            if (len(blocks) == 2 and
-                    all(order.name() == "degrevlex" for order in blocks)):
-                giac_order = "revlex"
-                var_names = var_names[:len(blocks[0])]
-            else:
-                raise NotImplementedError(
-                    "%s is not a supported term order in "
-                    "Giac Groebner bases." % P.term_order())
-
-        # compute de groebner basis with giac
-        gb_giac = F.gbasis(list(var_names), giac_order)
-
-    else:
-        gb_giac = F.eliminate(list(elim_variables), 'gbasis')
-
-    return PolynomialSequence(gb_giac, P, immutable=True)
+# sagemath_giac split its __init__.py into these two
```

Comment:
Not utterly important, but could we add a pytest that checks if this integration is working at least in principle by, say, calculating something simple.

Reply:
Can you elaborate?

In sagemath-giac, I have taken the contents of the old `__init__.py` and moved them into two separate files, `gb.py` and `context.py`. In sagelib I am importing those two modules inside of `__init__.py` so that, if sagemath-giac is installed, all of the old stuff will be available where you expect it using `from sage.libs.giac import foo`. If, on the other hand, sagemath-giac is not installed, then `import sage.libs.giac` will fail with an `ImportError`.

There are doctests that have `# needs sage.libs.giac` that will test this when sagemath-giac is installed. Or did you just want something for pytest specifically?

Reply:
I wanted a test that checks that these simple reexports work as expected (as a kind of silly integration test). But, as you write above, there are still tests in sage using giac, so this is not really needed.